### PR TITLE
Use dependabot for managing docker image upgrades

### PR DESCRIPTION
**Pull request by dannymidnight**
Configure dependabot as per this [helpful suggestion](https://github.com/buildkite/docs/pull/1904#issuecomment-1447543720).

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -9,3 +9,16 @@ updates:
     directory: "/vale"
     schedule:
       interval: daily
+  - package-ecosystem: "docker"
+    directory: "/"
+    schedule:
+      interval: "monthly"
+    reviewers:
+      - "buildkite/foundation"
```
Comment (yob) [inline .github/dependabot.yml:17]: Maybe onboarding?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -9,3 +9,16 @@ updates:
...
+    ignore:
+      # Permit dependabot to bump the digest only, we'll manually bump the ruby version when we're ready
+      - dependency-name: "*"
+        update-types:
+          - "version-update:semver-major"
+          - "version-update:semver-minor"
+          - "version-update:semver-patch"
```
Comment (yob) [inline .github/dependabot.yml:24]: fancy!

**Comment by ellsclytn** [comment]
Oh neat, you beat me to it 😄

**Comment by ellsclytn** [comment]
[PLT-456]

**Review by yob — APPROVED** [review]
Pre-approved, on the assumption dependabot PRs will ask onboarding for review instead of foundation 😅

File: .github/dependabot.yml (hunk 1 above)
Comment (dannymidnight) [reply .github/dependabot.yml:17]: hah yeah I wondered about this one. Onboarding sounds good 👍

**Pull request merged**